nsapass

```diff
@@ -178,6 +180,19 @@ def save(d):
             os.unlink(tempDB.name)
         tempDB.close()
 
+def lock_db():
+    dirname = os.path.dirname(DBLOCK)
+    os.makedirs(dirname, exist_ok=True)
+    try:
+        open(DBLOCK, 'xb').close()
+    except FileExistsError:
+        print_err('lock file "{}" already exists'.format(DBLOCK))
+        print_err('if no other instances of "nsapass" are running '
+            'remove the lock file manually')
+        print_err('exiting...')
+        sys.exit(1)
+    atexit.register(os.unlink, DBLOCK)
+
```

Comment:
This is another design of the locking function, using [atexit](https://docs.python.org/3/library/atexit.html) to release the lock.
I'm pretty happy with this: it keeps the cleanup in the vicinity of the code that takes the lock, avoids (visible) global variables and seems easier to understand than the `locked` modifier I came up with before.

Reply:
oo.  `atexit` neat. me _very_ like! 
ty ty.